src/transport/emitter.ts

```diff
@@ -131,4 +147,48 @@ export class Emitter {
     }
     return this.binaryEmitter(event, options);
   }
+
+  /**
+   * Return or create the Emitter singleton
+   *
+   * @return {Emitter} return Emitter singleton
+   */
+  static getInstance(): Emitter {
+    if (!Emitter.instance) {
+      Emitter.instance = new Emitter();
+    }
+    return Emitter.instance;
+  }
+
+  /**
+   * Add a listener for eventing
+   *
+   * @param {string} event type to listen to
+   * @param {Function} listener to call on event
+   * @return {void}
+   */
+  static on(event: "cloudevent" | "newListener" | "removeListener", listener: (...args: any[]) => void): void {
```

Comment:
I'm starting to lean towards allowing `any` in our linter. It makes me jump through ridiculous hoops sometimes to avoid it. In this case, I see you're using the same function signature as the Node.js `EventEmitter` type.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fbf9926cd51d7123e12cac280ccecaac01818bd3/types/node/events.d.ts#L63

We don't need to figure out how to get around this linting problem before this lands. Personally, I'd like to see the rule loosened, but I don't know TS linting rules well enough to loosen it only in certain situations (and I don't know what those would be). Maybe a warning is fine, but we'll be seeing this with every build from now on. :shrug:

Reply:
:+1: I'm onboard with you, I had to create crazy cast on some lines to please the linter, and EventEmitter is defined with any

Reply:
I added the change to this PR for @typescript-eslint/no-explicit-any